### DISCUSSION TITLE
The next/link router does not scroll to the top when clicked.

### DISCUSSION THREAD

**Discussion by singleseeker** [vercel/next.js#51207] [App Router] [answered]
![Kapture 2023-06-13 at 10 16 36](https://github.com/vercel/next.js/assets/520698/3e9a5b20-5b47-46d8-87b9-046f0464915f)

**Comment by devjiwonchoi** [comment]
Hmm.. that is weird. The default behavior for `Link` component is to scroll to the top.

> The default behavior of <Link> is to [scroll to the top of the route segment that has changed](https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#focus-and-scroll-management).

Have you configured any `href='/#hash'` or `scroll`?

```js
<Link href="/#hashid" scroll={false}>
  Scroll to specific id.
</Link>
```

**Reply by singleseeker** [reply]
Yep, it is weird.

Here is the structure. I've got a layout.js and categories/[slug]/page.js
<img width="326" alt="image" src="https://github.com/vercel/next.js/assets/520698/26fd76c6-7b7f-4525-81b3-6861978186be">

1. the layout.js file
```
import API from '@/utils/api';

import PageBanner from '../../components/PageBanner';
import Sidebar from '../components/Sidebar';


const Layout = async ({ children }) => {
  const responseCategories = await API.get(
    '/articles/categories?include=count.'
  );
  const categories = responseCategories.data;

  const responseTags = await API.get('/articles/tags?include=count');
  const tags = responseTags.data;

  return (
    <>
      <PageBanner>
        <>
          <h2 className="mt-1 mb-3">直码动态</h2>
          <p className="text-white m-0 p-0">
            了解行业趋势、掌握最新技术动向、追踪公司里程碑事件，直码动态让您尽览一切！
          </p>
        </>
      </PageBanner>

      <section className="blog-page-area py-4">
        <div className="container">
          <div className="row latest-article">
            <div className="col-lg-8 col-xl-9 col-md-12">{children}</div>

            <Sidebar tags={tags} categories={categories} />
          </div>
        </div>
      </section>
    </>
  );
};

export default Layout;
```

2. page.js
```
import API from '@/utils/api';

import BlogList from '../../../components/BlogList';
import BlogListPagination from '../../../components/Pagination';



export default async function Page({ searchParams, params }) {

  const { slug } = params;
  const { page } = searchParams || 1;

  const url = API.query()
    .for('articles')
    .params({ 'page[size]': 6, 'page[number]': page })
    .where('categories,' slug)
    .sort('-published_at')
    .url();

  const res = await API.get(url);
  const blogData = res.data;

  return (
    <>

      <h3 className="mb-3 title">最新文章</h3>

      <BlogList blogData={blogData} />

      <BlogListPagination meta={blogData.meta} />

    </>
  );
}
```

The result is here.
<img width="1606" alt="image" src="https://github.com/vercel/next.js/assets/520698/63f0d409-0b2c-459b-8353-5ea05786f9c6">

I'd like to share something new I've found. If the top line is shown in the view, you click other categories item; it would not scroll to the top. Also, that's why I am here to ask.

You can try it online by visiting: https://lab.zhimapro.com/blog/categories/9/

I think the reason is that the documentation said the next.js just helped me to focus and scroll to view the segment that's changed.
So here is another question, how could I stop this feature? 
<img width="1766" alt="image" src="https://github.com/vercel/next.js/assets/520698/b95e834b-0458-4ffb-8fc1-13939653ca86">

Just as you may see, if the web design has no sticky header, this feature is kind of cool.

But when you have a sticky header, it became weird.
<img width="1756" alt="image" src="https://github.com/vercel/next.js/assets/520698/6b957d70-8199-4b08-86f7-062135e89743">


> By default, Next.js will set focus and scroll into view the segment that's changed on navigation.

**Reply by devjiwonchoi** [reply]
How's your actual `Link` set up in your Navigation?

**Reply by singleseeker** [reply]
It is a usual way.

File categories.js
```
import Link from 'next/link'

export default function Categories({ categories }) {
  return (
    <div className="card">
      <div className="card-header">
        <h3 className="widget-title">分类</h3>
      </div>
      <div className="card-body">
        <ul className='categories'>
          {
            categories.data
              .filter(category => category.count.number > 0)
              .map((category) => (
                <li key={category.id}>
                  <Link href={`/blog/categories/${category.id}`}>
                    {category.name}
                    <span className="post-count">({category.count.number})</span>
                  </Link>
                </li>
              ))
          }
        </ul>
      </div>
    </div>
  )
}

```

**Reply by singleseeker** [reply]
By the way, if the top line is not shown on the view, everything works well.

**Reply by devjiwonchoi** [reply, marked as answer]
Ok, this seems like a known issue. #42492 #47475
Since this issue hasn't been resolved, I've brought an alternate solution from #42492.

```js
'use client'

import { useEffect } from 'react'

export default function ScrollUp() {
  useEffect(() => window.document.scrollingElement?.scrollTo(0, 0), [])

  return null
}
```

## Reference
- #42492 
- #47475

**Reply by singleseeker** [reply]
Thank you very much. It saved my life~ 
Actually, I did some research, but maybe the poor English, I've not got this info.
